include/mini/mesh/cgns/format.hpp

```diff
@@ -75,37 +73,38 @@ struct Coordinates {
   /**
    * Read coordinates from a given `(file, base, zone)` tuple.
    */
-  void Read(int file_id, int base_id, int zone_id) {
+  void Read() {
     // All id's are 1-based when passing to CGNS/MLL.
     cgsize_t first = 1;
     cgsize_t last = CountNodes();
     auto data_type = std::is_same_v<Real, double> ?
         CGNS_ENUMV(RealDouble) : CGNS_ENUMV(RealSingle);
-    cg_coord_read(file_id, base_id, zone_id, "CoordinateX",
+    cg_coord_read(file_id_, base_id_, zone_id_, "CoordinateX",
                   data_type, &first, &last, x_.data());
-    cg_coord_read(file_id, base_id, zone_id, "CoordinateY",
+    cg_coord_read(file_id_, base_id_, zone_id_, "CoordinateY",
                   data_type, &first, &last, y_.data());
-    cg_coord_read(file_id, base_id, zone_id, "CoordinateZ",
+    cg_coord_read(file_id_, base_id_, zone_id_, "CoordinateZ",
                   data_type, &first, &last, z_.data());
   }
   /**
    * Write coordinates to a given `(file, base, zone)` tuple.
    */
-  void Write(const int& file_id, const int& base_id, const int& zone_id) {
+  void Write() {
     int coord_id;
     auto data_type = std::is_same_v<Real, double> ?
         CGNS_ENUMV(RealDouble) : CGNS_ENUMV(RealSingle);
-    cg_coord_write(file_id, base_id, zone_id,
+    cg_coord_write(file_id_, base_id_, zone_id_,
                    data_type, "CoordinateX", x_.data(), &coord_id);
-    cg_coord_write(file_id, base_id, zone_id,
+    cg_coord_write(file_id_, base_id_, zone_id_,
                    data_type, "CoordinateY", y_.data(), &coord_id);
-    cg_coord_write(file_id, base_id, zone_id,
+    cg_coord_write(file_id_, base_id_, zone_id_,
                    data_type, "CoordinateZ", z_.data(), &coord_id);
   }
 
  private:  // Data Members:
   std::string name_;
   std::vector<Real> x_, y_, z_;
+  int file_id_, base_id_, zone_id_;
```

Comment:
I think it's better to save a pointer or reference to its parent.

Reply:
Got it.